api/src/main/java/org/openmrs/api/db/hibernate/HibernateAdministrationDAO.java

```diff
@@ -200,9 +202,51 @@ public void setApplicationContext(ApplicationContext applicationContext) throws
 	
 	/**
 	 * @see org.openmrs.api.db.AdministrationDAO#validate(java.lang.Object, Errors)
+	 * @should pass validation if all required fields have proper values
 	 */
 	@Override
 	public void validate(Object object, Errors errors) throws DAOException {
+		String name = object.getClass().getSimpleName();
+		Class entityClass = object.getClass();
+		for (Field field : object.getClass().getDeclaredFields()) {
+			field.setAccessible(true);
+			if (field.getName().equalsIgnoreCase(name)) {
+				Object value = null;
+				try {
+					value = field.get(object);
+				} catch (IllegalArgumentException e) {
+					log.warn("Unable to get the value of " + field.getName());
+				} catch (IllegalAccessException e) {
+					log.warn("Unable to get the value of " + field.getName());
+				}
+				if (value != null) {
+					int l1 = getMaximumPropertyLength(entityClass, field.getName());
+					int l2 = value.toString().length();
+					if (l2 > l1) {
+						errors.rejectValue(field.getName(),  "error.exceededMaxLengthOfField",new Object[] { l1 }, null);
```

Comment:
@lluismf please look into this, i have tried to use properties instead of hard coded message. 
thank you. :)